zappa_file_widget/url_widget.py

```diff
@@ -77,19 +77,23 @@ def render(self, name, value, attrs=None):
         attrs['type'] = 'hidden'
         parent_html = super(URLWidget, self).render(name, value, attrs=attrs)
         template = Template(html_template)
-        s3_host = 's3-ap-southeast-1.amazonaws.com'
-        default_region = getattr(settings, 'AWS_S3_HOST', s3_host).split(".amazonaws.com")[0].split("s3-")[1]
+        aws_access_key_id = getattr(settings, 'MEDIA_AWS_ACCESS_KEY_ID', 'AWS_ACCESS_KEY_ID')
+        aws_secret_access_key = getattr(settings, 'MEDIA_AWS_SECRET_ACCESS_KEY', 'AWS_SECRET_ACCESS_KEY')
+        s3_host_from_settings = getattr(settings, 'MEDIA_AWS_S3_HOST', 'AWS_S3_HOST')
+        s3_host = s3_host_from_settings or 's3-ap-southeast-1.amazonaws.com'
+        default_region = s3_host.split(".amazonaws.com")[0].split("s3-")[1]
+        aws_storage_bucket_name = getattr(settings, 'MEDIA_AWS_STORAGE_BUCKET_NAME', 'AWS_STORAGE_BUCKET_NAME')
```

Comment:
Excuse me if I do not understand your comment correctly: the proposed changes are meant to provide full backwards compatibility. Unless settings.MEDIA_xxx is provided the exact same settings.xxx your code uses is used for xxx. One example (the other two context keys behave in the exact same way):

1) branch **master**:
    context['AWS_ACCESS_KEY_ID'] == settings.AWS_ACCESS_KEY_ID

2) branch **feature/allow_media_differentiation**, with settings.MEDIA_AWS_ACCESS_KEY_ID **not defined**:
    context['AWS_ACCESS_KEY_ID'] == settings.AWS_ACCESS_KEY_ID

3) branch **feature/allow_media_differentiation**, with settings.MEDIA_AWS_ACCESS_KEY_ID **defined**:
    context['AWS_ACCESS_KEY_ID'] == settings.MEDIA_AWS_ACCESS_KEY_ID

case 2) defaults back to the same values as case 1) so someone upgrading to the new version of zappa-file-widget and not modifying his settings would see no difference.